File: FbxSharpTests/FbxAxisSystemTest.cs
using System;
using NUnit.Framework;
using FbxSharp;

namespace FbxSharpTests
{
    [TestFixture]
    public class FbxAxisSystemTest : TestBase
    {
        [Test]
        public void FbxAxisSystem_Create_HasDefaults()
        {
            // given:
            FbxAxisSystem obj;

            // when:
            obj = new FbxAxisSystem();

            // then:
            var sign = 0;
            Assert.That(obj.GetFrontVector(ref sign), Is.EqualTo(FbxAxisSystem.EFrontVector.eParityOdd));
            Assert.That(sign, Is.EqualTo(1));
            Assert.That(obj.GetUpVector(ref sign), Is.EqualTo(FbxAxisSystem.EUpVector.eYAxis));
            Assert.That(sign, Is.EqualTo(1));
            Assert.That(obj.GetCoorSystem(), Is.EqualTo(FbxAxisSystem.ECoordSystem.eRightHanded));
        }
    }
}

File: FbxCppTests/FbxAxisSystemTest.cpp

#include "Tests.h"

using namespace std;

void FbxAxisSystem_Create_HasDefaults()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxAxisSystem* obj;

    // when:
    obj = new FbxAxisSystem();

    // then:
    int sign = 0;
    AssertEqual(FbxAxisSystem::EFrontVector::eParityOdd, obj->GetFrontVector(sign));
    AssertEqual(1, sign);
    AssertEqual(FbxAxisSystem::EUpVector::eYAxis, obj->GetUpVector(sign));
    AssertEqual(1, sign);
    AssertEqual(FbxAxisSystem::ECoordSystem::eRightHanded, obj->GetCoorSystem());
}

void FbxAxisSystemTest::RegisterTestCases()
{
    AddTestCase(FbxAxisSystem_Create_HasDefaults);
}

